pkg/cableengine/cableengine.go

```diff
@@ -185,7 +200,12 @@ func (i *engine) InstallCable(endpoint types.SubmarinerEndpoint) error {
 		return nil
 	}
 
+	i.Lock()
+	i.natDiscoveryPending[endpoint.Spec.CableName]++
```

Comment:
Should this also handle the case where `i.natDiscoveryPending[endpoint.Spec.CableName]` is already >= 1?

Reply:
The intention of using a counter instead of simple bool (ie set) was to take into account possible multiple pending NAT discovery requests so we don't prematurely remove from `natDiscoveryPending` and miss a discovery response..

Reply:
OK, so doesn’t that mean we need to check the number of pending discoveries in line 227, instead of blindly removing `i.natDiscoveryPending[endpoint.Spec.CableName]`?

Reply:
In that case we don't want to process any pending discovery responses. That covers the edge case described in the summary.

Reply:
Oh yeah duh, the regular case is handled in line 141.